lib/lexer_mdx.mll

```diff
@@ -85,5 +85,6 @@ let token syntax lexbuf =
     match syntax with
     | Syntax.Normal -> text      None lexbuf
     | Syntax.Cram   -> cram_text None lexbuf
+    | Syntax.Mli    -> failwith "No lexer here for mli"
```

Comment:
What do you think about splitting this into `markdown_token` and `cram_token` so that we get rid of the failwith and properly dispatch between the various parsers in `Mdx` where we have all we need to parse every syntax?

Reply:
I'm not sure I know what you mean. Could you maybe sketch out what you're thinking this code would look like?

Reply:
What would you think if instead of having a:
```ocaml
val token : Syntax.t -> Lexing.lexbuf -> [...] list
```
we had:
```ocaml
val cram_token : Lexing.lexbuf -> [...] list
val markdown_token : Lexing.lexbuf -> [...] list
```
in `Lexer_mdx` and turned `Mdx.parse_lexbuf` into:
```ocaml
let parse_lexbuf file_contents syntax lexbuf =
  match syntax with
  | Mli -> Mli_parser.parse_mli file_contents
  | Normal -> parse (Lexer_mdx.markdown_token lexbuf)
  | Cram -> parse (Lexer_mdx.cram_token lexbuf)
```

So that we get rid of this unhandled `Mli` syntax in `Lexer_mdx`. We can clean it up further later but I'd rather get rid of that failwith since this `token` function is only used in one place, in which syntax is already known.

Reply:
@NathanReb Ah, that's a good idea -- this is what we do now.